pkg/ndctl/region.go

```diff
@@ -208,13 +204,25 @@ func (r *region) CreateNamespace(opts CreateNamespaceOpts) (Namespace, error) {
 	}
 
 	if opts.Size != 0 {
-		ways := uint64(C.ndctl_region_get_interleave_ways(r))
-		align = align * ways
-		if opts.Size%align != 0 {
-			// Round up size to align with next block boundary.
-			opts.Size = (opts.Size/align + 1) * align
-			klog.V(4).Infof("%s: namespace size must align to interleave-width:%d * alignment:%d, force-align to %d",
-				regionName, ways, align, opts.Size)
+		namespacealign := align * r.InterleaveWays()
+		regionalign := r.GetAlign()
+		// Size has to be aligned both by namespace alignment times interleave_ways, and also by region alignment
+		lcmalign := LCM(namespacealign, regionalign)
+		klog.V(3).Infof("%s: Least Common Multiple of namespacealign:%d and regionalign:%d is %d",
+			regionName, namespacealign, regionalign, lcmalign)
+		if opts.Size%regionalign != 0 {
```

Comment:
s/regionalign/lcmalign/ here.

Reply:
oops, leftover in replace, good catch, thanks!